src/index.js

```diff
@@ -22,19 +22,20 @@ class NearSocialViewerElement extends HTMLElement {
         const src = this.getAttribute('src');
         const code = this.getAttribute('code');
         const initialProps = this.getAttribute('initialProps');
+        const rpc = this.getAttribute('rpc');
 
         const root = createRoot(container);
-        root.render(<App src={src} code={code} initialProps={JSON.parse(initialProps)} selectorPromise={this.selectorPromise}/>);
+        root.render(<App src={src} code={code} rpc={rpc} initialProps={JSON.parse(initialProps)} selectorPromise={this.selectorPromise}/>);
     }
 
     static get observedAttributes() {
-        return ['src', 'code', 'initialProps'];
+        return ['src', 'code', 'initialProps', 'rpc'];
     }
 
     attributeChangedCallback(name, oldValue, newValue) {
         // Re-render the component with new props when attributes change
         if (oldValue !== newValue) {
-            const container = this.shadowRoot.querySelector('div');
+            const container = this.shadowRoot.querySelector('slot');
```

Comment:
What does this do?

Reply:
For some reason, it wasn't finding the div in the shadow root when it went to re-render. But querying for the slot picked up the node it was looking to render. It appeared to have the same effect. Not married to it, but it did enable the re-render to work (showing the rpc param as the attribute)